apps/aecore/lib/aecore/tx/data_tx.ex

```diff
@@ -38,80 +36,120 @@ defmodule Aecore.Tx.DataTx do
   ## Parameters
   - type: The type of transaction that may be added to the blockchain
   - payload: The strcuture of the specified transaction type
-  - sender: The public address of the account originating the transaction
+  - senders: The public addresses of the accounts originating the transaction
   - fee: The amount of tokens given to the miner
   - nonce: A random integer generated on initialisation of a transaction (must be unique!)
   """
-  defstruct [:type, :payload, :sender, :fee, :nonce]
+  defstruct [:type, :payload, :senders, :fee, :nonce]
   use ExConstructor
 
-  @spec init(tx_types(), payload(), binary(), integer(), integer()) :: DataTx.t()
-  def init(type, payload, sender, fee, nonce) do
-    %DataTx{type: type, payload: type.init(payload), sender: sender, fee: fee, nonce: nonce}
+  def valid_types do
+    [
+      Aecore.Account.Tx.SpendTx,
+      Aecore.Structures.CoinbaseTx,
+      Aecore.Oracle.Tx.OracleExtendTx,
+      Aecore.Oracle.Tx.OracleQueryTx,
+      Aecore.Oracle.Tx.OracleRegistrationTx,
+      Aecore.Oracle.Tx.OracleResponseTx
+    ]
+  end
+
+  @spec init(tx_types(), payload(), list(binary()) | binary(), non_neg_integer(), integer()) ::
+          DataTx.t()
+  def init(type, payload, senders, fee, nonce) when is_list(senders) do
+    %DataTx{type: type, payload: type.init(payload), senders: senders, nonce: nonce, fee: fee}
+  end
+
+  def init(type, payload, sender, fee, nonce) when is_binary(sender) do
+    %DataTx{type: type, payload: type.init(payload), senders: [sender], nonce: nonce, fee: fee}
+  end
+
+  def fee(%DataTx{fee: fee}) do
+    fee
+  end
+
+  def senders(%DataTx{senders: senders}) do
+    senders
+  end
+
+  def type(%DataTx{type: type}) do
+    type
+  end
+
+  def nonce(%DataTx{nonce: nonce}) do
+    nonce
+  end
+
+  def payload(%DataTx{payload: payload}) do
+    payload
+  end
+
+  def sender(tx) do
+    List.last(senders(tx))
```

Comment:
why should this be last, either rename or comment in code with explaination

Reply:
I will add more explanation and will consider changing the name from sender to "main_sender" or something like that.

Reply:
is the last the main? Isn't every sender equal in multisig? one is maybe the initiator

Reply:
Core team decided that there is one special sender whose nonce is updated. I decided to make the last one special. (Although I will probably change it to the first one for performance.)